src/main/java/com/google/sps/servlets/userServlet.java

```diff
@@ -10,13 +19,62 @@
 @WebServlet("/userinformation")
 public class userServlet extends HttpServlet {
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //dummy data for user
+    String username = "Sam";
+    
     Gson gson = new Gson();
-    MockUserData mockUserData = new MockUserData();
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("User");
+    
+    String userJson = "";
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for(Entity entity : results.asIterable()){
+      if(entity.getProperty("name") == username){
+        String email = (String) entity.getProperty("email");
+        String[] favouritesArr = gson.fromJson((String) entity.getProperty("favourites"), String[].class);
+        List<String> favourites = Arrays.asList(favouritesArr);
+        String[] pastReviewsArr = gson.fromJson((String) entity.getProperty("pastReviews"), String[].class);
+        List<String> pastReviews = Arrays.asList(pastReviewsArr);
 
-        response.setContentType("text/html;");
-        response.getWriter().println("<h1>Hello world!</h1>");
+        User user = new User(username, email, favourites, pastReviews);
+        userJson = String.format("{\"data\" : %s, \"id\": %s }", gson.toJson(user), entity.getKey().getId());
       }
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(userJson);
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    List<String> favourites = Arrays.asList("1","2","3","4");
+    List<String> pastReviews = Arrays.asList("5","6","7","8");
+
+    Entity user1 = new Entity("User");
+    user1.setProperty("name", "Sam");
+    user1.setProperty("email", "sam@gmail.com");
+    user1.setProperty("favourites", favourites);
+    user1.setProperty("pastReviews", pastReviews);
```

Comment:
It's not feasible to insert Java lists directly into Datastore! I had problems with this as well, you can use gson.toJson(favourites) to convert it into a json string before setting the property.